### ISSUE TITLE
[Due for payment 2026-03-19] [$250] Screen Readers: Many Pages: The focus is not moved when activating the link 'fix the errors'

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77337]
## Action Performed:
Precondition: The User should be logged in

1. Using Android TalkBack or iOS VoiceOver, open the Expensify app
2. Start the onboarding flow
3. Swipe to one of the options and double-tap to activate
4. Swipe to the 'Continue' button and double-tap to activate
5. Swipe to the 'fix the errors' link and double-tap
6. Note where the focus is moved.

Other Occurrences:
1. On Profile - Display Name
2. On Profile - Address
3. On Security - Close Account
4. On Wallet - Add Bank Account
5. On Save the World - I know a teacher

## Expected Result:
When the link is activated, the focus should move immediately and directly to the first input field that requires correction, allowing the user to begin fixing the form without losing context.

## Actual Result:
After the user taps the 'fix the errors' link, the keyboard focus remains on the link itself. The user is then forced to manually swipe through the entire form to find the first error field, making the error correction process inefficient.

## Area issue was found in:
Onboarding

## Failed WCAG checkpoints
WCAG 2.4.3 Focus Order
https://www.w3.org/WAI/WCAG22/Understanding/focus-order

## User impact:
People with mobility impairments who must rely on keyboard access for operating a page benefit from a logical, usable focus order. People with vision impairments or disabilities that make reading difficult can become disoriented when tabbing takes focus someplace unexpected or when they cannot easily find the content surrounding an interactive element. Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.

## Suggested resolution:
Ensure that upon activating the control, the focus is moved to the first field with an error.
https://appt.org/en/docs/react-native/samples/accessibility-focus

## Environment(s)
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos
![Image](https://github.com/user-attachments/assets/6e2fcbdf-d57e-40c4-8116-9c43c1cb5935)

https://github.com/user-attachments/assets/6126c5fc-dcab-4e12-afd2-a1967379ea52

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005704
**Issue reported by:** NA
**Slack conversation:** NA
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022016982792401313068</li>
        <li>Upwork Job ID: 2016982792401313068</li>
        <li>Last Price Increase: 2026-01-29</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by rushatgabhane** [comment]
Please feel free to use Melvin's analysis

**Comment by daledah** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After the user taps the 'fix the errors' link, the keyboard focus remains on the link itself. The user is then forced to manually swipe through the entire form to find the first error field, making the error correction process inefficient.

### What is the root cause of that problem?

When the "fix the errors" link is pressed, it triggers `onFixTheErrorsLinkPressed` in `FormWrapper.tsx` but this only moves keyboard/visual focus, NOT screen reader focus

https://github.com/Expensify/App/blob/bb2b54eecdf13807cd69040e305d76e38a91f490/src/components/Form/FormWrapper.tsx#L135-L137


### What changes do you think we should make in order to solve the problem?


We should call `Accessibility.moveAccessibilityFocus()` after `focusInput?.focus()` to move screen reader focus

```
    focusInput?.focus?.();
    Accessibility.moveAccessibilityFocus(focusInput)
```

https://github.com/Expensify/App/blob/bb2b54eecdf13807cd69040e305d76e38a91f490/src/components/Form/FormWrapper.tsx#L135-L137

### What alternative solutions did you explore? (Optional)


**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by daledah** [comment]
> @daledah can you please create a fix?

Sure, i'll create a follow-up PR soon

> If you’re not able to test the changes on iOS, I think it would be better to reassign it to someone else who can properly test it instead of blocking this.

As far as I know, the contributor has no way to test iOS via Ad Hoc.

**Comment by quinthar** [comment]
Approved $250 to @Krishna2323

**Comment by quinthar** [comment]
Approved $250 to @daledah
